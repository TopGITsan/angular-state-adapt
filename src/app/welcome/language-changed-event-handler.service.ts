import { inject, Injectable } from '@angular/core';
import { ApplicationEventHandler } from '@event-hub/event-handler/application-event-handler.interface';
import { LanguageChangedEvent } from '../events/language-changed.event';
import { WelcomeStoreService } from './welcome-store.service';

@Injectable()
export class LanguageChangedEventHandler implements ApplicationEventHandler {
  eventClass = LanguageChangedEvent;
  private readonly welcomeStore = inject(WelcomeStoreService);
  handle(event: LanguageChangedEvent) {
    console.log({ event });
    this.welcomeStore.language.next(event.lang);
  }
}
