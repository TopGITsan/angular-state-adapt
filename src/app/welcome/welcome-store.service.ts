import { Injectable } from '@angular/core';
import { Lang } from '@internalization/lang.types';
import { BehaviorSubject } from 'rxjs';

@Injectable({ providedIn: 'root' })
export class WelcomeStoreService {
  readonly language = new BehaviorSubject<Lang | null>(null);
  readonly selectedLanguage$ = this.language.asObservable();
}
