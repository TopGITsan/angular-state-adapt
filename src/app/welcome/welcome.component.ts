import { AsyncPipe, NgClass, TitleCasePipe } from '@angular/common';
import { ChangeDetectionStrategy, Component, inject } from '@angular/core';
import { TranslocoPipe } from '@jsverse/transloco';
import { WelcomeStoreService } from './welcome-store.service';

@Component({
  selector: 'app-welcome',
  imports: [TitleCasePipe, TranslocoPipe, AsyncPipe, NgClass],
  template: `
    <h1
      class="text-clamp3 block-transparente-clip-text font-vibes"
      [ngClass]="lang$ | async"
    >
      {{ 'hello' | transloco | titlecase }}!
    </h1>
  `,
  styles: `
    :host {
      @apply flex justify-center items-center h-full;
    }
    .block-transparente-clip-text {
      -webkit-text-stroke: 1px var(--mat-sys-on-surface);
      @apply inline-block text-transparent bg-clip-text;
    }

    .null {
      @apply bg-gradient-to-r from-white via-red-400 to-white;
    }

    .it {
      background-image: linear-gradient(
        90deg,
        #009246 0 33.33%,
        #ffffff 33.33% 66.66%,
        #ce2b37 66.66% 100%
      );
    }
    .ro {
      background-image: linear-gradient(
        90deg,
        #002b7f 0 33.33%,
        #fcd116 33.33% 66.66%,
        #ce1126 66.66% 100%
      );
    }
    .fr {
      background-image: linear-gradient(
        90deg,
        #002395 0 33.33%,
        #ffffff 33.33% 66.66%,
        #ed2939 66.66% 100%
      );
    }

    .es {
      background-image: linear-gradient(
        180deg,
        #f21830 0 40%,
        #ffd90d 40% 60%,
        #f21830 60% 100%
      );
    }
    .de {
      background-image: linear-gradient(
        180deg,
        #000 0 40%,
        #ff0000 40% 60%,
        #ffcc00 60% 100%
      );
    }

    .en {
      background-color: #fff;
      background-image: linear-gradient(
          180deg,
          transparent,
          transparent 45%,
          #c8102e 45%,
          #c8102e 55%,
          transparent 55%,
          transparent 100%
        ),
        linear-gradient(
          90deg,
          transparent,
          transparent 40%,
          #c8102e 40%,
          #c8102e 60%,
          transparent 60%,
          transparent 100%
        );
    }
  `,
  changeDetection: ChangeDetectionStrategy.OnPush,
})
export class WelcomeComponent {
  readonly lang$ = inject(WelcomeStoreService).selectedLanguage$;
}
