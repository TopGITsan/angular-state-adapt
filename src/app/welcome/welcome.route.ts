import { WelcomeComponent } from './welcome.component';

export const welcomePath = 'home';
export const WELCOME_ROUTE = {
  path: welcomePath,
  component: WelcomeComponent,
};
