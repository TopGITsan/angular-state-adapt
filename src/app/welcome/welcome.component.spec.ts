import { ComponentFixture, TestBed } from '@angular/core/testing';
import { getTranslocoTestingModule } from '@internalization/get-transloco-testing-module.function';
import { WelcomeComponent } from './welcome.component';

describe('HomeComponent', () => {
  let component: WelcomeComponent;
  let fixture: ComponentFixture<WelcomeComponent>;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      providers: [],
      imports: [WelcomeComponent, getTranslocoTestingModule()],
    }).compileComponents();

    fixture = TestBed.createComponent(WelcomeComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(fixture).toMatchSnapshot();
    expect(component).toBeTruthy();
  });

  it('should render title', () => {
    const fixture = TestBed.createComponent(WelcomeComponent);

    fixture.detectChanges();

    const compiled = fixture.nativeElement as HTMLElement;

    expect(compiled.querySelector('h1')?.textContent).toContain('Hello!');
  });
});
