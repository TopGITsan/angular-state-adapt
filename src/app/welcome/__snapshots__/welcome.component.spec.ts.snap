// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeComponent should create 1`] = `
<app-welcome
  lang$={[Function Observable]}
>
  <h1
    class="text-clamp3 block-transparente-clip-text font-vibes"
  >
     Hello! 
  </h1>
</app-welcome>
`;
